packages/ag-charts-community/src/module/optionsModule.ts

```diff
@@ -100,9 +102,22 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
         let options = deepClone(userOptions, cloneOptions);
 
         if (this.presetType != null) {
-            const presetOptions = (PRESETS as any)[this.presetType]?.(options, () => this.activeTheme) ?? options;
+            const presetConstructor: ((options: AgPresetOptions, activeTheme: () => ChartTheme) => T) | undefined = (
+                PRESETS as any
+            )[this.presetType];
+            const presetParams = options as any as AgPresetOptions;
+            const presetOptions = presetConstructor?.(presetParams, () => this.activeTheme) ?? options;
             this.debug('>>> AgCharts.createOrUpdate() - applying preset', options, presetOptions);
-            options = presetOptions;
+
+            const activeTheme = getChartTheme(options.theme);
+            const presetType = this.optionsType(presetOptions) as any as keyof AgPresetOverrides;
+            const presetTheme = activeTheme.presets[presetType];
+            if (presetTheme != null) {
+                const themedOptions = mergeDefaults(presetParams, presetTheme);
+                options = presetConstructor?.(themedOptions, () => this.activeTheme) ?? themedOptions;
```

Comment:
This is less than ideal - but to determine the theme, we need to know the preset type, so we need to run the preset constructor. Then if it has a theme, we need to run it again with the theme applied

Reply:
Do we really need to do this - for gauges it looks like `type` is required? For Financial Charts we could fallback to `presetType`?